core/src/main/java/org/apache/calcite/runtime/JsonFunctions.java

```diff
@@ -112,18 +113,22 @@ public static JsonPathContext jsonApiCommonSyntax(String input, String pathSpec)
   }
 
   public static JsonPathContext jsonApiCommonSyntax(JsonValueContext input, String pathSpec) {
+    PathMode mode;
+    String pathWff;
     try {
       Matcher matcher = JSON_PATH_BASE.matcher(pathSpec);
       if (!matcher.matches()) {
-        throw RESOURCE.illegalJsonPathSpec(pathSpec).ex();
+        mode = PathMode.STRICT;
+        pathWff = pathSpec;
```

Comment:
What if there is an illegal mode? For instance, json_exists('{\"foo\":\"bar\"}', "strictxx $.foo' false on error)". Will it throw exception?

Reply:
> false on error

The exception was handled and returned as false.